src/components/rangeslider/defaults.js

```diff
@@ -0,0 +1,38 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+
+
+module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
+
+    if(!layoutIn.xaxis || !layoutOut.xaxis) return;
+
+    var containerIn = layoutIn.xaxis.rangeslider || {},
+        containerOut = layoutOut.xaxis.rangeslider = {};
+
+    if(!containerIn.visible) return;
```

Comment:
use [`isPlainObject`](https://github.com/plotly/plotly.js/blob/master/src/lib/is_plain_object.js) to determine if the range slider is visible by default.